Providers/RustSyncManager.swift

```diff
@@ -0,0 +1,635 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Account
+import Shared
+import Storage
+import Sync
+import SyncTelemetry
+import AuthenticationServices
+import Common
+
+// Extends NSObject so we can use timers.
+public class RustSyncManager: NSObject, SyncManager {
+    // We shouldn't live beyond our containing BrowserProfile, either in the main app
+    // or in an extension.
+    // But it's possible that we'll finish a side-effect sync after we've ditched the
+    // profile as a whole, so we hold on to our Prefs, potentially for a little while
+    // longer. This is safe as a strong reference, because there's no cycle.
+    unowned fileprivate let profile: BrowserProfile
```

Comment:
```suggestion
    private let profile: BrowserProfile
```

Is there a reason you need `unowned` here? We use that mostly in closures inside the capture list `[unowned let...] in` but I am not sure what benefit we get for using unowned here. It might end up being dangerous because its a strong reference which has no relation to retain and release cycle. So you can see random crashes.

Reply:
No other reason that it was in [BrowserSyncManager](https://github.com/mozilla-mobile/firefox-ios/blob/7c90d922f0b2cdad3d311f6649ecc2c04d4e20f6/Providers/Profile.swift#L757).

Reply:
The interesting fact is that its an 8-year-old code that still works!

Personally, I avoid the `unowned` unless I am very confident in the lifecycle of the variable which might have been the case 8 years ago.